### ISSUE TITLE
wrong type when class inherit from `namedtuple`

### ISSUE THREAD

**Issue by NateScarlet** [microsoft/pyright#2325]
**Describe the bug**

instance of classes that inherit from namedtuple always has the namedtuple type, pyright show error when assign it to inherited type.

**To Reproduce**

Use code below.

**Expected behavior**

Got interited type instead of namedtuple type for instance.

**Screenshots or Code**

```python
from collections import namedtuple


Class1 = namedtuple("Class1", "name")


class Class2(Class1):
    some_class_member = 1


def new_class1() -> Class1:
    return Class2(name="a")


def new_class2() -> Class2:
    return Class2(name="a")


print(type(new_class2()))
```

![image](https://user-images.githubusercontent.com/21083014/134296280-9495769f-982c-4c20-a83a-e5b538ec80cb.png)

**VS Code extension or command-line**
VSCode extension: pylance v2021.9.2

**Additional context**

```
> py .\pyright_issue_reproduce.py
<class '__main__.Class2'>
```

**Comment by erictraut** [comment]
Thanks for the bug report. This will be fixed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.171 which was just published. It will also be included in the next release of pylance.
